scipy/optimize/_linprog_ip.py

```diff
@@ -0,0 +1,1940 @@
+            if i in no_adjust:
+                continue
+            lb, ub = b
+            if lb is None and ub is None:
+                n_unbounded += 1
+                x[i] = x[i] - x[n_x + n_unbounded - 1]
+            else:
+                if lb is None:
+                    x[i] = ub - x[i]
+                else:
+                    x[i] += lb
+
+    n_x = len(c)
+    x = x[:n_x]  # all the rest of the variables were artificial
+    fun = x.dot(c)
+    slack = b_ub - np.dot(A_ub, x)  # report slack for ORIGINAL UB constraints
+    # report residuals of ORIGINAL EQ constraints
+    con = b_eq - np.dot(A_eq, x)
+
+    if status == 0 and (np.isnan(x).any() or np.isnan(fun) or
+                        np.isnan(slack).any() or np.isnan(con).any()):
+        status = 4
+        message = "Numerical difficulties were encountered but no errors " \
+                  "were raised. This is known to occur if the 'presolve' " \
+                  "option is False, 'sparse' is True, and A_eq includes " \
+                  "redundant rows. If you encounter this under different " \
+                  "circumstances, please submit a bug report. Otherwise, " \
+                  "remove linearly dependent equations from your equality " \
+                  "constraints or enable presolve."
+
+    return x, fun, slack, con, status, message
+
+
+def _get_solver(sparse=False, lstsq=False, sym_pos=False, cholesky=False):
+    """
+    Given solver options, return a handle to the appropriate linear system
+    solver.
+
+    Parameters
+    ----------
+    sparse : boolean
+        True if the system to be solved is sparse. This is typically set
+        True when the original A_ub and A_eq matrices are sparse.
+    lstsq : boolean
+        True if the system is ill-conditioned and/or (nearly) singular and
+        thus a more robust least-squares solver is desired. This is sometimes
+        needed as the solution is approached.
+    sym_pos : boolean
+        True if the system matrix is symmetric positive definite
+        Sometimes this needs to be set false as the solution is approached,
+        even when the system be symmetric positive definite, due to numerical
+        difficulties.
+    cholesky : boolean
+        True if the system is to be solved by explicit Cholesky decomposition
+        followed by explicit forward/backward substitution. This is
+        occasionally faster for very large, symmetric positive definite
+        systems when a system with the same matrix is to be solved with
+        several right hand sides.
+
+    Returns
+    -------
+    solve : function
+        Handle to the appropriate solver function
+
+    """
+    if sparse:
+        if lstsq or sym_pos == False:
+            # solve = lambda M, r: sps.linalg.lsqr(M, r)[0]
+            def solve(M, r, sym_pos=False): 
+                return sps.linalg.lsqr(M, r)[0]
+#                return sps.linalg.lsqr(M, r, atol = 0, btol = 0, 
```

Comment:
From a round of `pep8` changes that `autopep8` couldn't handle (eliminating lambdas). K, fixed.